smarts/core/remote_agent.py

```diff
@@ -80,24 +82,36 @@ def __init__(
 
     def act(self, obs):
         """Call the agent's act function asynchronously and return a Future."""
-        self._act_future = self._worker_stub.act.future(
+        self._grpc_future = self._worker_stub.act.future(
             worker_pb2.Observation(payload=cloudpickle.dumps(obs))
         )
 
-        return self._act_future
+        def get_action_future():
+            action = cloudpickle.loads(
+                        self._grpc_future.result().action
+                    )   
+            return action
+         
+        future = self._action_executor.submit(get_action_future)
+
+        return future
 
     def start(self, agent_spec: AgentSpec):
         """Send the AgentSpec to the agent runner."""
         # Cloudpickle used only for the agent_spec to allow for serialization of lambdas.
+        self._action_executor = futures.ThreadPoolExecutor(max_workers=1)
```

Comment:
I have some thought that this `Executor` should be owned by the `RemoteAgentBuffer` and assigned to each `RemoteAgent` instance by passing the reference through the initialiser.